pkg/apiserver/filters/audit.go

```diff
@@ -96,27 +96,26 @@ func WithAudit(handler http.Handler, attributeGetter RequestAttributeGetter, out
 			internalError(w, req, err)
 			return
 		}
+
+		groups := auditStringSlice(attribs.GetUser().GetGroups())
+
 		asuser := req.Header.Get(authenticationapi.ImpersonateUserHeader)
 		if len(asuser) == 0 {
 			asuser = "<self>"
 		}
 		asgroups := "<lookup>"
 		requestedGroups := req.Header[authenticationapi.ImpersonateGroupHeader]
 		if len(requestedGroups) > 0 {
-			quotedGroups := make([]string, len(requestedGroups))
-			for i, group := range requestedGroups {
-				quotedGroups[i] = fmt.Sprintf("%q", group)
-			}
-			asgroups = strings.Join(quotedGroups, ", ")
+			asgroups = auditStringSlice(requestedGroups)
 		}
 		namespace := attribs.GetNamespace()
 		if len(namespace) == 0 {
 			namespace = "<none>"
 		}
 		id := uuid.NewRandom().String()
 
-		line := fmt.Sprintf("%s AUDIT: id=%q ip=%q method=%q user=%q as=%q asgroups=%q namespace=%q uri=%q\n",
-			time.Now().Format(time.RFC3339Nano), id, utilnet.GetClientIP(req), req.Method, attribs.GetUser().GetName(), asuser, asgroups, namespace, req.URL)
+		line := fmt.Sprintf("%s AUDIT: id=%q ip=%q method=%q user=%q groups=%q as=%q asgroups=%q namespace=%q uri=%q\n",
```

Comment:
> Which means that with foo"bar as the only group you will get "\"foo\\\"bar\"" as output?

That is the result I expect.  One parsing rule.



> Compare also the test-case below: would have expectedgroups=instead ofgroups=""`.

Also the expected result.  It's a quoted string of quoted strings separated by spaces.  Same as `asgroups` before it.